src/caffe/layers/mvn_layer.cu

```diff
@@ -113,7 +113,12 @@ void MVNLayer<Dtype>::Backward_gpu(const vector<Blob<Dtype>*>& top,
 
     caffe_gpu_div(temp_.count(), bottom_diff, temp_.gpu_data(), bottom_diff);
   } else {
-    caffe_copy(temp_.count(), top_diff, bottom_diff);
+    caffe_gpu_gemv<Dtype>(CblasNoTrans, num, dim, 1. / dim, top_diff,
+            sum_multiplier_.gpu_data(), 0., mean_.mutable_gpu_diff());
+    caffe_gpu_gemm<Dtype>(CblasNoTrans, CblasNoTrans, num, dim, 1, -1.,
+            mean_.gpu_diff(), sum_multiplier_.gpu_data(), 0.,
+            temp_.mutable_gpu_diff());
+    caffe_gpu_add(temp_.count(), top_diff, temp_.gpu_diff(), bottom_diff);
```

Comment:
Is there a reason to use the `mean_` and `temp_` Blobs' `diff`s rather than their `data` here, as in the CPU implementation? If not, `data` should be used to save memory.